spec/person_spec.rb

```diff
@@ -0,0 +1,45 @@
+require_relative '../person'
+
+describe Person do
+  context 'Creating a new person' do
+    it 'has a name and age' do
+      person = Person.new(25, 'John')
+      expect(person.name).to eq('John')
+      expect(person.age).to eq(25)
+    end
+
+    it 'defaults to an unknown name' do
+      person = Person.new(25)
+      expect(person.name).to eq('Unknown')
+    end
+
+    it 'can use services if of age' do
+      person = Person.new(25)
+      expect(person.can_use_services?).to be_truthy
+    end
+
+    it 'can use services if parent permission is given' do
+      person = Person.new(10, 'Jane', true)
+      expect(person.can_use_services?).to be_truthy
+    end
+
+    it 'cannot use services without parent permission if underage' do
+      person = Person.new(10, 'Jane', false)
+      expect(person.can_use_services?).to be_falsey
+    end
+
+    it 'generates a random id between 1 and 1000' do
+      person = Person.new(25)
+      expect(person.id).to be_between(1, 1000).inclusive
+    end
+
+    it 'adds a rental to their rentals list' do
+      person = Person.new(25)
+      book = double('book')
+      rental = double('rental')
+      allow(Rental).to receive(:new).and_return(rental)
+      person.add_rental('2023-03-16', book)
+      expect(person.rentals).to include(rental)
+    end
```

Comment:
- [ ] When I test this block of code, I get this error. Kindly fix it
<img width="735" alt="Screenshot 2023-03-17 at 08 59 55" src="https://user-images.githubusercontent.com/64774936/225846962-e2091996-1bb2-4143-8e28-4ad25d1eb8ca.png">

Reply:
Hello @Ghiftee, can you please share more insight to this, like what codeline did you use to check this because it passes all the test at my end. Maybe I am not doing something right.
![image](https://user-images.githubusercontent.com/105938633/225857612-bf87cdd0-e82b-449e-9885-0d0ac039fd1f.png)